pkg/overlay/client.go

```diff
@@ -51,7 +51,7 @@ type Options struct {
 
 // NewClient returns a new intialized Overlay Client
 func NewClient(identity *provider.FullIdentity, address string) (Client, error) {
-	tc := transport.NewClient(identity)
+	tc := transport.NewClient(identity, &Cache{}) // add overlay to transport client as observer
```

Comment:
there is not a cache created elsewhere that we would want to use? If not, this is fine.

Reply:
We dont' have access to a cache created elsewhere at this level, without getting import cycle errors. Besides, it only needs access to that function.